### ISSUE TITLE
Forgiving behaviour when trying to connect to MySQL

### ISSUE THREAD

**Issue by Badestrand** [brianc/node-postgres#2111]
Hi!

_Edit: Sorry, I was stupid and tried to connect to a MySQL instance.. However, I still think it should not crash the Node application but rather just emit an error. So please do not just close but consider the fix, thank you! Also I adjusted the title.._

First off, thanks for the great work.

Second, I encountered a bug (I think). When connecting to a certain Google Cloud connection my Node.js crashes with
```
/xyz/node_modules/pg/lib/connection.js:130
      var eventName = msg.name === 'error' ? 'errorMessage' : msg.name
                          ^

TypeError: Cannot read property 'name' of undefined
    at Socket.<anonymous> (/xyz/node_modules/pg/lib/connection.js:130:27)
    at Socket.emit (events.js:188:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
[nodemon] app crashed - waiting for file changes before starting...
```
That occurs [here](https://github.com/brianc/node-postgres/blob/master/packages/pg/lib/connection.js#L130).

I checked and it seems the `Connection.parseMessage` function (see [here](https://github.com/brianc/node-postgres/blob/master/packages/pg/lib/connection.js#L453)) returns undefined, resulting in this crash.

When the crash occurs the packet is `<Buffer 35 2e 37 2e 32 35>` or ASCII "5.7.25", if that matters.

My noobie suggestion for a fix would be to have
```
default:
      throw new Error('Corrupt response in communication with server')
```
in `Connection.parseMessage` function (see [here](https://github.com/brianc/node-postgres/blob/master/packages/pg/lib/connection.js#L453)) and then wrap things in `Connection.attachListeners` (see [here](https://github.com/brianc/node-postgres/blob/master/packages/pg/lib/connection.js#L130)) in a try-catch:
```
Connection.prototype.attachListeners = function (stream) {
  var self = this
  stream.on('data', function (buff) {
    try {
      // ...
    }
    catch (err) {
      self.emit('error', err)
    }
  })
  stream.on('end', function () {
    self.emit('end')
  })
}
```

**Comment by charmander** [comment]
Duplicate of #1940

**Comment by Pleximar** [comment]
Same error, does it works with a other port?
